### PR TITLE
Add example ingressroute objects

### PR DESCRIPTION
**Pull request by rosskukulinski**
As part of #450, I wanted to create put together a suite of IngressRoute objects and corresponding services.

This should hopefully be helpful for the development team working on the IngressRoute DAG implementations.

### CODE REVIEW

File: deployment/example-workload/ingressroute/01-ingress-to-ingressroute/name-based.ingressroute.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#   name: name-example
+# spec:
+#   rules:
+#   - host: foo1.bar.com
+#     http:
+#       paths:
+#       - backend:
+#           serviceName: s1
+#           servicePort: 80
+#   - host: bar1.foo.com
```
Comment (alexbrand) [inline deployment/example-workload/ingressroute/01-ingress-to-ingressroute/name-based.ingressroute.yaml]: should this be `bar1.bar.com`?

File: deployment/example-workload/ingressroute/02-basic/default-health-checks.ingressroute.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+metadata: 
+  name: health-check
+  namespace: default
+spec: 
+  virtualhost:
+    fqdn: health.bar.com
+  routes: 
+    - match: /
+      services: 
+        - name: s1-def-health
```
Comment (alexbrand) [inline deployment/example-workload/ingressroute/02-basic/default-health-checks.ingressroute.yaml]: Do we need to specify the healthchecks somewhere?

**Comment by rosskukulinski** [comment]
Good catch, @alexbrand. updated

File: deployment/example-workload/ingressroute/00-common/01-services.yaml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+spec:
+  selector:
+    app: kuard
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+---
+---
```
Comment (davecheney) [inline deployment/example-workload/ingressroute/00-common/01-services.yaml]: s/---//

**Comment by alexbrand** [comment]
@rosskukulinski This LGTM, but needs a rebase if you don't mind.

**Comment by rosskukulinski** [comment]
@alexbrand rebased

**Pull request merged**